pkg/types/intoto/v0.0.1/entry.go

```diff
@@ -219,10 +228,26 @@ func (v *V001Entry) validate() error {
 	// TODO handle multiple
 	pk := v.keyObj.(*x509.PublicKey)
 
-	// This also gets called in the CLI, where we won't have this data
+	// one of two cases must be true:
+	// - ProposedEntry: client gives an envelope and NOT hash/payloadhash (to be computed server-side) OR
+	// - CommittedEntry: NO envelope and hash/payloadHash must be present
 	if v.IntotoObj.Content.Envelope == "" {
+		if v.IntotoObj.Content.Hash == nil {
+			return fmt.Errorf("missing hash value for envelope")
+		} else if err := v.IntotoObj.Content.Hash.Validate(strfmt.Default); err != nil {
+			return fmt.Errorf("validation error on envelope hash: %w", err)
+		}
+		if v.IntotoObj.Content.PayloadHash == nil {
+			return fmt.Errorf("missing hash value for payload")
+		} else if err := v.IntotoObj.Content.PayloadHash.Validate(strfmt.Default); err != nil {
+			return fmt.Errorf("validation error on payload hash: %w", err)
+		}
+		// if there is no envelope, and hash/payloadHash are valid, then there's nothing else to do here
 		return nil
+	} else if v.IntotoObj.Content.Hash != nil || v.IntotoObj.Content.PayloadHash != nil {
```

Comment:
Any concern with this being a breaking change, if clients are currently sending these values?

Reply:
great point; let me think through that.

Reply:
you're right, this is probably too strict for an already released type.